### PR TITLE
Updates to bash completion for `dockerd`

### PR DESCRIPTION
**Pull request by albers**
A bunch of options were missing in bash completion of `dockerd`.
See individual commit messages for details and references.
ping @sdurrheimer for zsh completion

### CODE REVIEW

File: contrib/completion/bash/docker [hunk 1]
```diff
@@ -2124,10 +2147,20 @@ _docker_daemon() {
 			__docker_complete_log_options
 			return
 			;;
+		--metrics-addr)
+			__docker_complete_local_interfaces --ip-only
```
Comment (thaJeztah) [inline contrib/completion/bash/docker]: Wonder if it would be "cleaner" to have a separate function for ip-addresses

Comment (albers) [inline contrib/completion/bash/docker]: @thaJeztah updated, thanks for the hint.

**Comment by vieux** [comment]
LGTM

**Comment by vieux** [comment]
LGTM

**Pull request merged**